python_ta/transforms/ExprWrapper.py

```diff
@@ -0,0 +1,139 @@
+from typing import Optional
+
+import astroid
+import z3
+from astroid import nodes
+
+
+class ExprWrapper:
+    node: astroid.NodeNG
+    types = {}
+
+    def __init__(self, expr: nodes.Expr, types={}):
+        self.node = expr.value
+        self.types = types
+        self.transformed = self.reduce(self.node)
+
+    def reduce(self, node: astroid.NodeNG) -> Optional[z3.ExprRef]:
+        """Convert astroid node to z3 expression and return it.
+        If an error is encountered or a case is not considered, return None."""
+        try:
+            if isinstance(node, nodes.BoolOp):
+                node = self.parse_bool_op(node)
+            elif isinstance(node, nodes.UnaryOp):
+                node = self.parse_unary_op(node)
+            elif isinstance(node, nodes.Compare):
+                node = self.parse_compare(node)
+            elif isinstance(node, nodes.BinOp):
+                node = self.parse_bin_op(node)
+            elif isinstance(node, nodes.Const):
+                node = node.value
+            elif isinstance(node, nodes.Name):
+                node = self.apply_name(node.name, self.types[node.name])
+            else:
+                node = None
+
+            return node
+        except Exception:
+            return None
+
+    def apply_name(self, name: str, typ: str) -> Optional[z3.ExprRef]:
+        """Set up the appropriate variable representation in Z3 based on name and type.
+        If an error is encountered or a case is unconsidered, return None."""
+        # TODO: determine full list of supported types
+        if typ == "int":
+            return z3.Int(name)
+        if typ == "float":
+            return z3.Real(name)
+        if typ == "bool":
+            return z3.Bool(name)
+
+        return None
+
+    def parse_compare(self, node: astroid.Compare) -> Optional[z3.ExprRef]:
+        left, ops = node.left, node.ops
+        # ERROR: left, right can be None at any point in the execution
```

Comment:
Right, so using exceptions will remove this comment. (Basically we're assuming that it's just `self.reduce` and `self.apply_bin_op` that raise errors, but this method itself doesn't *assuming* that the others don't. This method can't introduce new errors.)